### DISCUSSION TITLE
Adding context to an Alert

### DISCUSSION THREAD

**Discussion by KeeganOP** [grafana/grafana#47442] [Alerting]
Currently Alerts lack the context in which the Alert occured. For example, say I wanted to alert on the percentage failure rate of an endpoint using loki. 

Loglines:
```
{"endpoint": "/users", "response_code": "200"}
{"endpoint": "/users", "response_code": "200"}
{"endpoint": "/users", "response_code": "500"}
{"endpoint": "/users", "response_code": "500"}
{"endpoint": "/users", "response_code": "200"}
{"endpoint": "/users", "response_code": "200"}
{"endpoint": "/users", "response_code": "500"}
{"endpoint": "/users", "response_code": "200"}
{"endpoint": "/users", "response_code": "500"}
{"endpoint": "/users", "response_code": "200"}
```

Easy to calculate in a panel (take the fail count divided by the total count * 100):
```
sum by (endpoint) (count_over_time({label="value3"} | json | response_code!="200" [1m])) / on (endpoint) sum by (endpoint) (count_over_time({label="value3"} | json [1m])) * 100
```

We can now build an alert to fire if the value of this query is above say 20% (so more than 20% of the calls are failing). I'm going to use legacy alerts here but the idea stays the same even with grafana 8 alerts:
![image](https://user-images.githubusercontent.com/87812012/162197723-81d79256-a255-48e7-9de3-8c0043d2fa59.png)

![image](https://user-images.githubusercontent.com/87812012/162196673-b63af12b-06ce-499e-b595-2c57340eadd7.png)

We now have the alert. This is cool and all but I have no way of knowing if the above 20% failure rate is because of  1 out of 2 calls failing (50% error rate) or if its from 10000 out of 20000 calls failing (also 50% failure rate but way more important). The current alerting works to alert but I dont have the scale or context of that alert. 

When looking into this problem I started looking at the message templates and grafana 8 alerts. Thinking that I could have 2 panels (one that reports the error count `$A` and one that reports the total count `$B`). Add an expression to math the two panels (`$C`) and alert on that, thinking that I would be able to access the results of `$values.A` and `$values.B` in the template. Until I saw that the message template only offers the values from the panel that was evaluated to trigger the alert (`$C`). Not all the panels in the alert. 

I find myself now having no way of adding context to an alert. If my SLO's are 98% if a low volume endpoint gets called and fails in a non critical way Im forced to look at every alert even if that endpoint is only called twice in a 3 min period. 

Workaround I've thought about is adding a reduce and only alerting if the count of calls is above a specific number. And then im missing the context of the %... my alert will then be "endpoint has an above 20% failure rate" but i wont have the % number nor the total / error number

All this could be solved with grafana 8 alerts allowing me to access all the values in all the panels (`$values.A`, `$values.B`, `$values.C` from the example above). My condition would then be C is above 20% and my alert message:
```
"{{ $labels.A.endpoint }} has a failure rate of {{ $values.C }}%. {{ $values.A }} calls failed out of {{ $values.B }}"
```
With render being:
```
"/users has a failure rate of 33.3%. 50 calls failed out of 150"
```

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
